src/proof/mod.rs

```diff
@@ -149,12 +149,15 @@ impl FoldingMode {
 }
 
 /// A trait for a prover that works with a field `F`.
-pub trait Prover<'a, F: CurveCycleEquipped, M: FrameLike<Ptr, FrameIO = Vec<Ptr>>> {
+pub trait Prover<'a, F: CurveCycleEquipped> {
+    /// Associated type for a frame-like datatype
+    type M: FrameLike<Ptr, FrameIO = Vec<Ptr>>;
```

Comment:
Can we have a more distinctive name other than "M"?

Reply:
Suggestions? I think M came from Multiframe, but it is outdated indeed. I guess Frame can work

Reply:
Yeah, `Frame` or `FrameType` would work I think